hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsBlobClient.java

```diff
@@ -435,6 +434,11 @@ public AbfsRestOperation createPath(final String path,
               null);
         }
       }
+      Path parentPath = new Path(path).getParent();
+      if (parentPath != null && !parentPath.isRoot()) {
+        createMarkers(parentPath, overwrite, permissions, isAppendBlob, eTag,
+            contextEncryptionAdapter, tracingContext, isNamespaceEnabled);
+      }
     }
     if (isFile) {
```

Comment:
Nit: There is a condition few lines below which checks if its not a file. These 2 can be combined as if else construct.

Reply:
taken